src/libraries/System.Private.CoreLib/src/System/Int128.cs

```diff
@@ -1290,22 +1290,33 @@ public static Int128 Clamp(Int128 value, Int128 min, Int128 max)
         /// <inheritdoc cref="INumber{TSelf}.CopySign(TSelf, TSelf)" />
         public static Int128 CopySign(Int128 value, Int128 sign)
         {
-            Int128 absValue = value;
-
-            if (IsNegative(absValue))
-            {
-                absValue = -absValue;
-            }
-
-            if (IsPositive(sign))
+#if TARGET_32BIT
+            Int128 t = value;
+            if (Int128.IsPositive(sign))
             {
-                if (IsNegative(absValue))
+                if (Int128.IsNegative(t))
                 {
-                    Math.ThrowNegateTwosCompOverflow();
+                    t = -t;
+                    if (Int128.IsNegative(t))
+                    {
+                        Math.ThrowNegateTwosCompOverflow();
+                    }
                 }
-                return absValue;
+                return t;
             }
-            return -absValue;
+            if (Int128.IsPositive(t))
+            {
+                t = -t;
+            }
+            return t;
+#else
+            // Int128.MinValue == value && 0 <= sign
+            if ((long)value._upper == long.MinValue && value._lower == 0 && (long)sign._upper >= 0)
```

Comment:
I'd just do this the simple/readable way as:
```suggestion
            if ((value == Int128.MinValue) && IsPositive(sign))
```

The JIT should take care of inlining the comparison and IsPositive check to give good codegen.